### PR TITLE
Feature/Permissions, Template, and Views Cleanup [LEI-435]

### PR DESCRIPTION
**Pull request by pattisdr**
# Ticket
https://openscience.atlassian.net/browse/LEI-435

# Purpose

Auditing  all views, making sure right data is displaying and correct permissions are in place. Cleaning up templates, responsiveness, adding pagination, fixing broken things, remove obsolete views/code.

In Progress:
(Experimenter)
- [x] Study List 
- [x] Create Study 
- [x] Study detail
- [x] Edit Study
- [x] Build Study (modify json)
- [x] View Study Responses
- [x] Researcher List
- [x] Researcher Detail
- [x] Participant List
- [x] Participant Detail
- [x] Navbar

(Lookit)
- [x] Study List
- [x] Study Detail
- [x] Participant Profile
- [x] Add Children
- [x] Add demographic data
- [x] Update email preferences
- [x] Remove/Update children

### CODE REVIEW

File: web/views.py [hunk 1]
```diff
@@ -272,17 +147,178 @@ def post(self, request, *args, **kwargs):
...
+        in the URLconf, but subclasses can override this to return any object.
+        '''
+        if queryset is None:
+            queryset = self.get_queryset()
+        # ChildUpdate View needs to be called with slug or pk - uuid in URLconf
+        # instead so use this to lookup child
+        uuid = self.kwargs.get('uuid')
+
+        if uuid is not None:
+            queryset = queryset.filter(uuid=uuid)
```
Comment (cwisecarver) [inline web/views.py]: [This](https://docs.djangoproject.com/en/1.11/topics/http/shortcuts/#get-object-or-404) can cut out some boilerplate.

File: web/views.py [hunk 2]
```diff
@@ -272,17 +147,178 @@ def post(self, request, *args, **kwargs):
...
+            raise Http404(_('No %(verbose_name)s found matching the query') %
+                          {'verbose_name': queryset.model._meta.verbose_name})
+        return obj
+
+    def get_context_data(self, **kwargs):
+        '''
+        If authenticated, add demographic presence, and children to context data dict
+        '''
+        context = super().get_context_data(**kwargs)
+        context['user'] = self.request.user
```
Comment (cwisecarver) [inline web/views.py]: We're using the `django.template.context_processors.request` context processor so the request should already be accessible in any template as `request`.

**Comment by pattisdr** [comment]
Thanks @cwisecarver, addressing now

**Comment by pattisdr** [comment]
Comments addressed @cwisecarver

**Pull request merged**